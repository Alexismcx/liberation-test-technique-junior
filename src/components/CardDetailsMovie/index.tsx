import {
  Card,
  useTheme,
  CardContent,
  Typography,
  Rating,
  Box,
} from "@mui/material";

export interface ICardDetailsMovie {
  title: string;
  id: number;
  overview: string;
  releaseDate: string;
  posterPath: string;
  listRatedMovies: { id: number; value: number }[];
  handleChangeRating: (id: number, value: number) => void;
}

const CardDetailsMovie: React.FC<ICardDetailsMovie> = ({
  title,
  id,
  overview,
  releaseDate,
  posterPath,
  listRatedMovies,
  handleChangeRating,
}) => {
  const theme = useTheme();
  return (
    <Card sx={{ maxWidth: 500, width: "900px" }}>
      <Box display="flex" position="relative">
        <Box
          display="flex"
          justifyContent="center"
          alignItems="center"
          flexDirection="column"
          sx={{
            height: 350,
            width: "100%",
            objectFit: "cover",
            background: `linear-gradient(rgba(0, 75, 170, 0.5), rgba(0,0,0,0)), url('https://image.tmdb.org/t/p/w500/${posterPath}')`,
          }}
        >
          <Typography
            variant="h2"
            color={theme.palette.text.secondary}
            textTransform="uppercase"
            textAlign="center"
          >
            {title}
          </Typography>
          <Rating
            name="size-medium"
            defaultValue={0}
            value={listRatedMovies.find((movie) => movie.id === id)?.value || 0}
            onChange={(e, newValue) => handleChangeRating(newValue, id)}
          />
        </Box>
      </Box>
      <CardContent>
        <Box display="flex" justifyContent="center" gap={2}>
          <Typography variant="body1">{releaseDate}</Typography>
        </Box>
        <Typography
          textAlign="center"
          color={theme.palette.text.primary}
          variant="body2"
        >
          {overview}
        </Typography>
      </CardContent>
    </Card>
  );
};

export default CardDetailsMovie;
