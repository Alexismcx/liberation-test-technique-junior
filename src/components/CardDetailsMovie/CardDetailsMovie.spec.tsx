import { fireEvent, render, screen } from "@testing-library/react";
import CardDetailsMovie from ".";

const mockMovie = {
  title: "Title Detail",
  id: 1,
  overview: "Test overview",
  releaseDate: "2023-10-19",
  posterPath: "/test-poster.jpg",
  handleClickDetails: jest.fn(),
  handleAddFavorite: jest.fn(),
  handleChangeRating: jest.fn(),
  listRatedMovies: [],
  listFavoritedMovies: [],
};

describe(`<CardDetailsMovie>`, () => {
  beforeEach(() => {
    render(<CardDetailsMovie {...mockMovie} />);
  });
  it("renders correctly CardMovie component", () => {
    const titleElement = screen.getByText("Title Detail");
    expect(titleElement).toBeTruthy();
  });
  it("should handles rating change", () => {
    const ratingComponent = screen.getByTestId("rating-star");
    fireEvent.change(ratingComponent, {
      target: { value: 3 },
    });
    expect(mockMovie.handleChangeRating).toHaveBeenCalledWith(3, 1);
  });
  it("should match the snapshot", () => {
    const titleElement = screen.getByText("Title Detail");
    expect(titleElement).toMatchSnapshot();
  });
});
