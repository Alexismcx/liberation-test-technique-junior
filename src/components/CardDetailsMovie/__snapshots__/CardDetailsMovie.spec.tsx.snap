// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardDetailsMovie> should match the snapshot 1`] = `
<h2
  class="MuiTypography-root MuiTypography-h2 css-2982by-MuiTypography-root"
>
  Title Detail
</h2>
`;
